crates/router/src/core/payments/helpers.rs

```diff
@@ -1812,15 +1816,76 @@ pub async fn retrieve_card_with_permanent_token(
             message: "no customer id provided for the payment".to_string(),
         })?;
+
+    if business_profile.is_network_tokenization_enabled {
+        if let Some(mandate_ids) = mandate_id {
+            if let Some(api_models::payments::MandateReferenceId::NetworkTokenWithNTI(nt_data)) =
+                mandate_ids.mandate_reference_id
+            {
+                if let Some(ref pm_data) = payment_method_info {
+                    if let Some(network_token_locker_id) = &pm_data.network_token_locker_id {
+                        let mut token_data = cards::get_card_from_locker(
+                            state,
+                            customer_id,
+                            &payment_intent.merchant_id,
+                            network_token_locker_id.as_ref(),
+                        )
+                        .await
+                        .change_context(errors::ApiErrorResponse::InternalServerError)
+                        .attach_printable(
+                            "failed to fetch network token information from the permanent locker",
+                        )?;
+                        let expiry = nt_data.token_exp_month.zip(nt_data.token_exp_year);
+                        if let Some((exp_month, exp_year)) = expiry {
+                            token_data.card_exp_month = exp_month;
+                            token_data.card_exp_year = exp_year;
+                        }
+                        let network_token_data = domain::NetworkTokenData {
+                            token_number: token_data.card_number,
+                            token_cryptogram: None,
+                            token_exp_month: token_data.card_exp_month,
+                            token_exp_year: token_data.card_exp_year,
+                            nick_name: token_data.nick_name.map(masking::Secret::new),
+                            card_issuer: None,
+                            card_network: None,
+                            card_type: None,
+                            card_issuing_country: None,
+                            bank_code: None,
+                        };
+                        return Ok(domain::PaymentMethodData::NetworkToken(network_token_data));
+                    }
+                }
+            }
+        }
+
+        if let Some(ref pm_data) = payment_method_info {
+            if let Some(token_ref) = pm_data.network_token_requestor_reference_id.clone() {
+                let network_token_data = network_tokenization::get_token_from_tokenization_service(
+                    state,
+                    merchant_key_store,
+                    token_ref,
+                    pm_data,
+                )
+                .await
+                .change_context(errors::ApiErrorResponse::InternalServerError)
+                .attach_printable("failed to fetch network token data from tokenization service");
+                if let Ok(network_token_data) = network_token_data {
```

Comment:
Why the error case is discarded here?

Reply:
if any error from network tokenization, we should not stop the payment flow. error should not be propagated.